### PR TITLE
fix(aws): Configure aws provider

### PR DESCRIPTION
**Pull request by lwander**
Corrects assumptions about how clouddriver/front50 handle credentials & also ensures .aws/credentials file is generated & put in the right place when necessary.

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -1030,9 +1032,8 @@ hal config provider aws account add ACCOUNT [parameters]
 ```
 #### Parameters
 `ACCOUNT`: The name of the account to operate on.
- * `--access-key-id, --access-key`: Your AWS Access Key ID. If not provided, Halyard will try to find AWS as described at http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
  * `--account-id`: (*Required*) Your AWS account ID to manage. See http://docs.aws.amazon.com/IAM/latest/UserGuide/console_account-alias.html for more information.
- * `--assume-role, --role`: If set, Halyard will configure a credentials provider that use AWS Security Token Service to assume the specified role.
+ * `--assume-role`: (*Required*) If set, Halyard will configure a credentials provider that use AWS Security Token Service to assume the specified role.
```
Comment (duftler) [inline docs/commands.md]: s/use/uses

File: docs/commands.md [hunk 2]
```diff
@@ -1071,11 +1071,10 @@ hal config provider aws account edit ACCOUNT [parameters]
 ```
 #### Parameters
 `ACCOUNT`: The name of the account to operate on.
- * `--access-key-id, --access-key`: Your AWS Access Key ID. If not provided, Halyard will try to find AWS as described at http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
  * `--account-id`: Your AWS account ID to manage. See http://docs.aws.amazon.com/IAM/latest/UserGuide/console_account-alias.html for more information.
  * `--add-region`: Add this region to the list of managed regions.
  * `--add-required-group-membership`: Add this group to the list of required group memberships.
- * `--assume-role, --role`: If set, Halyard will configure a credentials provider that use AWS Security Token Service to assume the specified role.
+ * `--assume-role`: If set, Halyard will configure a credentials provider that use AWS Security Token Service to assume the specified role.
```
Comment (duftler) [inline docs/commands.md]: s/use/uses

File: docs/commands.md [hunk 3]
```diff
@@ -3106,7 +3118,7 @@ Edit configuration for the "s3" persistent store.
 hal config storage s3 edit [parameters]
 ```
 #### Parameters
- * `--access-key-id, --access-key`: Your AWS Access Key ID. If not provided, Halyard will try to find AWS as described at http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
+ * `--access-key-id, --access-key`: Your AWS Access Key ID. If not provided, Halyard/Spinnaker will try to find AWS credentials as described at http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
```
Comment (duftler) [inline docs/commands.md]: Looks like `--access-key` is no longer relevant?

File: docs/commands.md [hunk 4]
```diff
@@ -3106,7 +3118,7 @@ Edit configuration for the "s3" persistent store.
...
  * `--assume-role, --role`: If set, Halyard will configure a credentials provider that use AWS Security Token Service to assume the specified role.
```
Comment (duftler) [inline docs/commands.md]: s/use/uses

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/aws/AwsProvider.java [hunk 5]
```diff
@@ -20,8 +20,32 @@
...
+
+@Data
 public class AwsProvider extends Provider<AwsAccount> {
+  private String accessKeyId;
+  private String secretAccessKey;
+
+  private String defaultAssumeRole;
+  private String defaultKeyPairTemplate = "{{name}}-keypair";
+
+  private List<AwsRegion> defaultRegions = Collections.singletonList(new AwsRegion().setName("us-west-2"));
```
Comment (duftler) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/aws/AwsProvider.java]: Can this default region be moved to an externalized config setting, or at least to a final constant up above?

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/aws/AwsAccountValidator.java [hunk 6]
```diff
@@ -44,6 +45,7 @@
   public void validate(ConfigProblemSetBuilder p, AwsAccount n) {
     DaemonTaskHandler.message(String.format("Validating AWS account named %s with %s", n.getNodeName(), getClass().getSimpleName()));
 
+    /*
```
Comment (duftler) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/aws/AwsAccountValidator.java]: Can this commented-out block be removed?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesDistributedService.java [hunk 7]
```diff
@@ -153,11 +153,13 @@ default String getArtifactId(String deploymentName) {
     ServiceSettings thisServiceSettings = resolvedConfiguration.getServiceSettings(thisService);
     SpinnakerRuntimeSettings runtimeSettings = resolvedConfiguration.getRuntimeSettings();
     Integer version = getRunningServiceDetails(details, runtimeSettings).getLatestEnabledVersion();
+    System.out.println("latest version " + version);
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesDistributedService.java]: Extraneous println?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesDistributedService.java [hunk 8]
```diff
@@ -153,11 +153,13 @@ default String getArtifactId(String deploymentName) {
...
     if (version == null) {
       version = 0;
     } else {
       version++;
     }
+    System.out.println("next version " + version);
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesDistributedService.java]: Extraneous println?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/google/GoogleFront50Service.java [hunk 9]
```diff
@@ -42,6 +43,14 @@
   GoogleDistributedServiceDelegate googleDistributedServiceDelegate;
 
   @Override
+  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
+    generateAwsProfile(deploymentConfiguration, endpoints, "/home/spinnaker").ifPresent(p -> profiles.add(p));
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/google/GoogleFront50Service.java]: Move to config or a constant?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesClouddriverService.java [hunk 10]
```diff
@@ -41,6 +45,13 @@ public DistributedLogCollector getLogCollector() {
   }
 
   @Override
+  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
+    generateAwsProfile(deploymentConfiguration, endpoints, "/root").ifPresent(p -> profiles.add(p));
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesClouddriverService.java]: Move to config or a constant?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesFront50Service.java [hunk 11]
```diff
@@ -41,6 +45,13 @@ public DistributedLogCollector getLogCollector() {
   }
 
   @Override
+  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
+    generateAwsProfile(deploymentConfiguration, endpoints, "/root").ifPresent(p -> profiles.add(p));
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesFront50Service.java]: Move to config or a constant?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/debian/LocalDebianClouddriverService.java [hunk 12]
```diff
@@ -48,6 +52,13 @@ LogCollector getLocalLogCollector() {
   }
 
   @Override
+  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
+    generateAwsProfile(deploymentConfiguration, endpoints, "/home/spinnaker").ifPresent(p -> profiles.add(p));
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/debian/LocalDebianClouddriverService.java]: Move to config or a constant?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/debian/LocalDebianFront50Service.java [hunk 13]
```diff
@@ -48,6 +52,13 @@ LogCollector getLocalLogCollector() {
   }
 
   @Override
+  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
+    generateAwsProfile(deploymentConfiguration, endpoints, "/home/spinnaker").ifPresent(p -> profiles.add(p));
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/debian/LocalDebianFront50Service.java]: Move to config or a constant?

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/aws/AwsProvider.java (hunk 5 above)
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/aws/AwsProvider.java]: In a sense this is the externalized config - this only comes into play the first time your halconfig is generated after installing, and this is the only place this constant is referenced.

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/aws/AwsAccountValidator.java (hunk 6 above)
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/aws/AwsAccountValidator.java]: Ah let me fix this

**Pull request merged**